Svelte 5: Unit testing of $derived fails

### Describe the bug

I am writing a unit test for a function that returns a `$derived()` rune, but it doesn't seem to get updated. This is the function I am testing:

```ts
// lib/test.svelte.ts
export function test(initial: number) {
   let num = $state(initial);
   const double = $derived(num * 2);

   return {
      get value() { return num; },
      get double() { return double; },
      addOne() { num += 1; }
    };
}
```

and this is the unit test

```ts
// index.test.ts
import { describe, it, expect } from 'vitest';
import { test } from './lib/test.svelte';

describe('test runes', () => {
    it('works', async () => {
        const t = test(42);
        expect(t.value).toBe(42);
        expect(t.double).toBe(84);
        t.addOne();
        expect(t.value).toBe(43);
        expect(t.double).toBe(86);
    })
})
```

### Reproduction

Here's a [StackBlitz](https://stackblitz.com/edit/sveltejs-kit-template-default-2nsbtq?file=src%2Findex.test.ts,src%2Froutes%2F%2Bpage.svelte) showing the problem. The runes work fine in the rendered HTML, but when you run `npm test`, you get this





### Logs

```shell
FAIL  src/index.test.ts > test runes > works
AssertionError: expected 84 to be 86 // Object.is equality

- Expected
+ Received

- 86
+ 84

 ❯ eval src/index.test.ts:17:20
     15|   t.addOne();
     16|   expect(t.value).toBe(43);
     17|   expect(t.double).toBe(86);
       |                    ^
     18|  });
     19| });
```
```


### System Info

```shell
System:
    OS: Linux 5.0 undefined
    CPU: (2) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.18.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm
    pnpm: 8.14.0 - /usr/local/bin/pnpm
  npmPackages:
    svelte: 5.0.0-next.35 => 5.0.0-next.35
```


### Severity

annoyance

--- a/documentation/docs/05-misc/02-testing.md
+++ b/documentation/docs/05-misc/02-testing.md
@@ -2,7 +2,155 @@
 title: Testing
 ---
 
-- component testing basics
-- rune testing basics
-- vitest setup
-- e2e
+Testing helps you write and maintain your code and guard against regressions. Testing frameworks help you with that, allowing you to describe assertions or expectations about how your code should behave. Svelte is unopinionated about which testing framework you use, so you can write unit tests, integration tests, and end-to-end tests using solutions like Vitest, Jasmine, Cypress and Playwright.
+
+## Unit/Integration-testing using Vitest
+
+Unit tests allow you to test small isolated parts of your code. Integration tests allow you to test parts of your application to see if they work together. We recommend using [Vitest](https://vitest.dev/) for these kinds of tests (this assumes you use Vite for bundling your application).
+
+To get started, install Vitest (`npm install vitest -D`) and JSDOM (a library that shims DOM APIs, `npm install jsdom -D`). Then adjust your `vite.config.js` such that you can simulate a browser environment for testing Svelte components on the client:
+
+```js
+/// file: vitest.config.js
+import { defineConfig } from 'vitest/config';
+
+export default defineConfig({
+	plugins: [
+		/* ... */
+	],
+	test: {
+		// If you are testing components client-side, you need to setup a DOM environment.
+		// If not all your files should have this environment, you can use a
+		// `// @vitest-environment jsdom` comment at the top of the test files instead.
+		environment: 'jsdom'
+	},
+	// Tell Vitest to prefer the browser condition. Svelte uses a so-called exports map
+	// which loads different code depending on the environment - we want to load the browser environment.
+	resolve: process.env.VITEST
+		? {
+				conditions: ['browser']
+			}
+		: undefined
+});
+```
+
+After that, you can create a test file in which you import the component to test, then interact with it programmatically and write expectations about the results:
+
+```js
+/// file: component.test.js
+import { flushSync, mount, unmount } from "svelte";
+import { expect, test } from "vitest";
+import Component from "./Component.svelte";
+
+test("Component", () => {
+	// Instantiate the component using Svelte's mount API
+	const comp = mount(Component, {
+		target: document.body, // document.body is available thanks to JSDOM
+		props: { initial: 0 },
+	});
+
+	expect(document.body.innerHTML).toBe("<button>0</button>");
+
+	// Click the button, then flush the changes so you can synchronously write expectations
+	document.body.querySelector("button")!.click();
+	flushSync();
+
+	expect(document.body.innerHTML).toBe("<button>1</button>");
+
+	// Remove the component from the DOM
+	unmount(comp);
+});
+
+```
+
+While the process is very straightforward, it is also low level and somewhat brittle - the structure of your component might change and you then may need to adjust your query selectors more often than not. Solutions like [@testing-library/svelte](https://testing-library.com/docs/svelte-testing-library/intro/) can help you facilitate this problem.
+
+### Using runes inside your test files
+
+It is possible to use runes inside your test files. First ensure your bundler knows to route the file through the Svelte compiler before running the test by adding `.svelte` to the filename (e.g `counter.svelte.test.js`). After that, you can use runes inside your tests.
+
+```js
+/// file: counter.svelte.test.js
+import { flushSync, mount, unmount } from 'svelte';
+import { expect, test } from 'vitest';
+
+test('Counter', () => {
+	let count = $state(0);
+	let double = $derived(count * 2);
+
+	expect(count).toEqual(0);
+	expect(double).toEqual(0);
+
+	count = 5;
+
+	expect(count).toEqual(5);
+	expect(double).toEqual(10);
+});
+```
+
+If the tested code uses effects, you need to wrap the test inside `$effect.root` to create a scope in which these are properly captured.
+
+```js
+/// file: counter.svelte.test.js
+import { flushSync, mount, unmount } from 'svelte';
+import { expect, test } from 'vitest';
+
+test('Effect', () => {
+	const cleanup = $effect.root(() => {
+		let count = $state(0);
+		let high = false;
+
+		$effect(() => {
+			if (count > 10) {
+				high = true;
+			}
+		});
+
+		// effects normally run after a microtask,
+		// use flushSync to execute all pending effects synchronously
+		flushSync();
+		expect(high).toEqual(false);
+
+		count = 15;
+		flushSync();
+
+		expect(high).toEqual(true);
+	});
+
+	cleanup();
+});
+```
+
+## E2E tests using Playwright
+
+E2E (short for "end to end") tests allow you to test your full application through the eyes of the user. This section uses [Playwright](https://playwright.dev/) as an example, but you can also use other solutions like [Cypress](https://www.cypress.io/) or [NightwatchJS](https://nightwatchjs.org/).
+
+To get start with Playwright, either let you guide by [their VS Code extension](https://playwright.dev/docs/getting-started-vscode), or install it from the command line using `npm init playwright`. It is also part of the setup CLI when you run `npm create svelte`.
+
+After you've done that, you should have a `tests` folder and a playwright config. You may need to adjust that config to tell Playwright what to do before running the tests - mainly starting your application at a certain port:
+
+```js
+/// file: playwright.config.js
+const config = {
+	webServer: {
+		command: 'npm run build && npm run preview',
+		port: 4173
+	},
+	testDir: 'tests',
+	testMatch: /(.+\.)?(test|spec)\.[jt]s/
+};
+
+export default config;
+```
+
+You can now start writing tests. These are totally unaware of Svelte as a framework, so you mainly interact with the DOM and write assertions.
+
+```js
+/// file: tests/hello-world.spec.js
+import { expect, test } from '@playwright/test';
+
+test('home page has expected h1', async ({ page }) => {
+	await page.goto('/');
+	await expect(page.locator('h1')).toBeVisible();
+});
+```

--- a/documentation/docs/05-misc/02-testing.md
+++ b/documentation/docs/05-misc/02-testing.md
@@ -44,24 +44,24 @@ After that, you can create a test file in which you import the component to test
 
 ```js
 /// file: component.test.js
-import { flushSync, mount, unmount } from "svelte";
-import { expect, test } from "vitest";
-import Component from "./Component.svelte";
+import { flushSync, mount, unmount } from 'svelte';
+import { expect, test } from 'vitest';
+import Component from './Component.svelte';
 
-test("Component", () => {
+test('Component', () => {
 	// Instantiate the component using Svelte's mount API
 	const comp = mount(Component, {
 		target: document.body, // document.body is available thanks to JSDOM
 		props: { initial: 0 },
 	});
 
-	expect(document.body.innerHTML).toBe("<button>0</button>");
+	expect(document.body.innerHTML).toBe('<button>0</button>');
 
 	// Click the button, then flush the changes so you can synchronously write expectations
-	document.body.querySelector("button")!.click();
+	document.body.querySelector('button')!.click();
 	flushSync();
 
-	expect(document.body.innerHTML).toBe("<button>1</button>");
+	expect(document.body.innerHTML).toBe('<button>1</button>');
 
 	// Remove the component from the DOM
 	unmount(comp);
@@ -124,7 +124,7 @@ test('Effect', () => {
 
 ## E2E tests using Playwright
 
-E2E (short for "end to end") tests allow you to test your full application through the eyes of the user. This section uses [Playwright](https://playwright.dev/) as an example, but you can also use other solutions like [Cypress](https://www.cypress.io/) or [NightwatchJS](https://nightwatchjs.org/).
+E2E (short for 'end to end') tests allow you to test your full application through the eyes of the user. This section uses [Playwright](https://playwright.dev/) as an example, but you can also use other solutions like [Cypress](https://www.cypress.io/) or [NightwatchJS](https://nightwatchjs.org/).
 
 To get start with Playwright, either let you guide by [their VS Code extension](https://playwright.dev/docs/getting-started-vscode), or install it from the command line using `npm init playwright`. It is also part of the setup CLI when you run `npm create svelte`.
 

--- a/documentation/docs/05-misc/02-testing.md
+++ b/documentation/docs/05-misc/02-testing.md
@@ -4,7 +4,7 @@ title: Testing
 
 Testing helps you write and maintain your code and guard against regressions. Testing frameworks help you with that, allowing you to describe assertions or expectations about how your code should behave. Svelte is unopinionated about which testing framework you use, so you can write unit tests, integration tests, and end-to-end tests using solutions like Vitest, Jasmine, Cypress and Playwright.
 
-## Unit/Integration testing using Vitest
+## Unit/Integration tests using Vitest
 
 Unit tests allow you to test small isolated parts of your code. Integration tests allow you to test parts of your application to see if they work together. If you're using Vite (including via SvelteKit), we recommend using [Vitest](https://vitest.dev/).
 
@@ -17,7 +17,7 @@ npm install -D vitest jsdom
 Then adjust your `vite.config.js`:
 
 ```js
-/// file: vitest.config.js
+/// file: vite.config.js
 import { defineConfig } from 'vitest/config';
 
 export default defineConfig({

--- a/documentation/docs/05-misc/02-testing.md
+++ b/documentation/docs/05-misc/02-testing.md
@@ -8,10 +8,102 @@ Testing helps you write and maintain your code and guard against regressions. Te
 
 Unit tests allow you to test small isolated parts of your code. Integration tests allow you to test parts of your application to see if they work together. If you're using Vite (including via SvelteKit), we recommend using [Vitest](https://vitest.dev/).
 
-To get started, install Vitest and jsdom (a library that shims DOM APIs):
+To get started, install Vitest:
 
 ```bash
-npm install -D vitest jsdom
+npm install -D vitest
+```
+
+Then adjust your `vite.config.js`:
+
+```diff
+/// file: vite.config.js
+- import { defineConfig } from 'vite';
++ import { defineConfig } from 'vitest/config';
+
+export default defineConfig({ /* ... */ })
+```
+
+You can now write unit tests for code inside your `.js/.ts` files:
+
+```js
+/// file: counter.svelte.test.js
+import { flushSync } from 'svelte';
+import { expect, test } from 'vitest';
+import { multiplier } from './multiplier.js';
+
+test('Multiplier', () => {
+	let double = multiplier(0, 2);
+
+	expect(double.value).toEqual(0);
+
+	double.set(5);
+
+	expect(double.value).toEqual(10);
+});
+```
+
+### Using runes inside your test files
+
+It is possible to use runes inside your test files. First ensure your bundler knows to route the file through the Svelte compiler before running the test by adding `.svelte` to the filename (e.g `multiplier.svelte.test.js`). After that, you can use runes inside your tests.
+
+```js
+/// file: counter.svelte.test.js
+import { flushSync } from 'svelte';
+import { expect, test } from 'vitest';
+import { multiplier } from './multiplier.svelte.js';
+
+test('Multiplier', () => {
+	let count = $state(0);
+	let double = multiplier(() => count, 2);
+
+	expect(double.value).toEqual(0);
+
+	count = 5;
+
+	expect(double.value).toEqual(10);
+});
+```
+
+If the tested code uses effects, you need to wrap the test inside `$effect.root` to create a scope in which these are properly captured.
+
+```js
+/// file: logger.svelte.test.js
+import { flushSync } from 'svelte';
+import { expect, test } from 'vitest';
+import { logger } from './logger.svelte.js';
+
+test('Effect', () => {
+	const cleanup = $effect.root(() => {
+		let count = $state(0);
+		// logger uses an $effect to log updates of its input
+		let log = logger(() => count);
+
+		// effects normally run after a microtask,
+		// use flushSync to execute all pending effects synchronously
+		flushSync();
+		expect(log.value).toEqual([0]);
+
+		count = 1;
+		flushSync();
+
+		expect(log.value).toEqual([0, 1]);
+	});
+
+	cleanup();
+});
+```
+
+### Component testing
+
+It is possible to test your components in isolation using Vitest.
+
+> Before writing component tests, think about whether you actually need to test the component, or if it's more about the logic _inside_ the component. If so, consider extracting out that logic to test it in isolation, without the overhead of a component
+
+To get started, install jsdom (a library that shims DOM APIs):
+
+```bash
+npm install -D jsdom
 ```
 
 Then adjust your `vite.config.js`:
@@ -90,59 +182,6 @@ test('Component', async () => {
 ```
 
 When writing component tests that test two-way-bindings, context or snippet props, it's best to create a wrapper component for your specific test and interact with that. `@testing-library/svelte` contains some [examples](https://testing-library.com/docs/svelte-testing-library/example) on how to achieve that.
-
-> If you find yourself writing lots of component tests, think about whether you actually need to test the component, or if it's more about the logic _inside_ the component. If so, consider extracting out that logic to test it in isolation, without the overhead of a component
-
-### Using runes inside your test files
-
-It is possible to use runes inside your test files. First ensure your bundler knows to route the file through the Svelte compiler before running the test by adding `.svelte` to the filename (e.g `multiplier.svelte.test.js`). After that, you can use runes inside your tests.
-
-```js
-/// file: counter.svelte.test.js
-import { flushSync, mount, unmount } from 'svelte';
-import { expect, test } from 'vitest';
-import { multiplier } from './multiplier.svelte.js';
-
-test('Multiplier', () => {
-	let count = $state(0);
-	let double = multiplier(() => count, 2);
-
-	expect(double.value).toEqual(0);
-
-	count = 5;
-
-	expect(double.value).toEqual(10);
-});
-```
-
-If the tested code uses effects, you need to wrap the test inside `$effect.root` to create a scope in which these are properly captured.
-
-```js
-/// file: logger.svelte.test.js
-import { flushSync, mount, unmount } from 'svelte';
-import { expect, test } from 'vitest';
-import { logger } from './logger.svelte.js';
-
-test('Effect', () => {
-	const cleanup = $effect.root(() => {
-		let count = $state(0);
-		// logger uses an $effect to log updates of its input
-		let log = logger(() => count);
-
-		// effects normally run after a microtask,
-		// use flushSync to execute all pending effects synchronously
-		flushSync();
-		expect(log.value).toEqual([0]);
-
-		count = 1;
-		flushSync();
-
-		expect(log.value).toEqual([0, 1]);
-	});
-
-	cleanup();
-});
-```
 
 ## E2E tests using Playwright
 

--- a/documentation/docs/05-misc/02-testing.md
+++ b/documentation/docs/05-misc/02-testing.md
@@ -2,7 +2,7 @@
 title: Testing
 ---
 
-Testing helps you write and maintain your code and guard against regressions. Testing frameworks help you with that, allowing you to describe assertions or expectations about how your code should behave. Svelte is unopinionated about which testing framework you use — you can write unit tests, integration tests, and end-to-end tests using solutions like Vitest, Jasmine, Cypress and Playwright.
+Testing helps you write and maintain your code and guard against regressions. Testing frameworks help you with that, allowing you to describe assertions or expectations about how your code should behave. Svelte is unopinionated about which testing framework you use — you can write unit tests, integration tests, and end-to-end tests using solutions like [Vitest](https://vitest.dev/), [Jasmine](https://jasmine.github.io/), [Cypress](https://www.cypress.io/) and [Playwright](https://playwright.dev/).
 
 ## Unit and integration testing using Vitest
 
@@ -27,7 +27,7 @@ export default defineConfig({ /* ... */ })
 You can now write unit tests for code inside your `.js/.ts` files:
 
 ```js
-/// file: counter.svelte.test.js
+/// file: multiplier.svelte.test.js
 import { flushSync } from 'svelte';
 import { expect, test } from 'vitest';
 import { multiplier } from './multiplier.js';
@@ -48,7 +48,7 @@ test('Multiplier', () => {
 It is possible to use runes inside your test files. First ensure your bundler knows to route the file through the Svelte compiler before running the test by adding `.svelte` to the filename (e.g `multiplier.svelte.test.js`). After that, you can use runes inside your tests.
 
 ```js
-/// file: counter.svelte.test.js
+/// file: multiplier.svelte.test.js
 import { flushSync } from 'svelte';
 import { expect, test } from 'vitest';
 import { multiplier } from './multiplier.svelte.js';
@@ -65,7 +65,7 @@ test('Multiplier', () => {
 });
 ```
 
-If the tested code uses effects, you need to wrap the test inside `$effect.root` to create a scope in which these are properly captured.
+If the code being tested uses effects, you need to wrap the test inside `$effect.root`:
 
 ```js
 /// file: logger.svelte.test.js
@@ -181,7 +181,7 @@ test('Component', async () => {
 });
 ```
 
-When writing component tests that test two-way-bindings, context or snippet props, it's best to create a wrapper component for your specific test and interact with that. `@testing-library/svelte` contains some [examples](https://testing-library.com/docs/svelte-testing-library/example) on how to achieve that.
+When writing component tests that involve two-way bindings, context or snippet props, it's best to create a wrapper component for your specific test and interact with that. `@testing-library/svelte` contains some [examples](https://testing-library.com/docs/svelte-testing-library/example).
 
 ## E2E tests using Playwright
 

--- a/documentation/docs/05-misc/02-testing.md
+++ b/documentation/docs/05-misc/02-testing.md
@@ -76,6 +76,7 @@ import { logger } from './logger.svelte.js';
 test('Effect', () => {
 	const cleanup = $effect.root(() => {
 		let count = $state(0);
+
 		// logger uses an $effect to log updates of its input
 		let log = logger(() => count);
 

--- a/documentation/docs/05-misc/02-testing.md
+++ b/documentation/docs/05-misc/02-testing.md
@@ -67,54 +67,49 @@ While the process is very straightforward, it is also low level and somewhat bri
 
 ### Using runes inside your test files
 
-It is possible to use runes inside your test files. First ensure your bundler knows to route the file through the Svelte compiler before running the test by adding `.svelte` to the filename (e.g `counter.svelte.test.js`). After that, you can use runes inside your tests.
+It is possible to use runes inside your test files. First ensure your bundler knows to route the file through the Svelte compiler before running the test by adding `.svelte` to the filename (e.g `multiplier.svelte.test.js`). After that, you can use runes inside your tests.
 
 ```js
 /// file: counter.svelte.test.js
 import { flushSync, mount, unmount } from 'svelte';
 import { expect, test } from 'vitest';
+import { multiplier } from './multiplier.svelte.js';
 
-test('Counter', () => {
+test('Multiplier', () => {
 	let count = $state(0);
-	let double = $derived(count * 2);
+	let double = multiplier(() => count, 2);
 
-	expect(count).toEqual(0);
-	expect(double).toEqual(0);
+	expect(double.value).toEqual(0);
 
 	count = 5;
 
-	expect(count).toEqual(5);
-	expect(double).toEqual(10);
+	expect(double.value).toEqual(10);
 });
 ```
 
 If the tested code uses effects, you need to wrap the test inside `$effect.root` to create a scope in which these are properly captured.
 
 ```js
-/// file: counter.svelte.test.js
+/// file: logger.svelte.test.js
 import { flushSync, mount, unmount } from 'svelte';
 import { expect, test } from 'vitest';
+import { logger } from './logger.svelte.js';
 
 test('Effect', () => {
 	const cleanup = $effect.root(() => {
 		let count = $state(0);
-		let high = false;
-
-		$effect(() => {
-			if (count > 10) {
-				high = true;
-			}
-		});
+		// logger uses an $effect to log updates of its input
+		let log = logger(() => count);
 
 		// effects normally run after a microtask,
 		// use flushSync to execute all pending effects synchronously
 		flushSync();
-		expect(high).toEqual(false);
+		expect(log.value).toEqual([0]);
 
-		count = 15;
+		count = 1;
 		flushSync();
 
-		expect(high).toEqual(true);
+		expect(log.value).toEqual([0, 1]);
 	});
 
 	cleanup();

--- a/documentation/docs/05-misc/02-testing.md
+++ b/documentation/docs/05-misc/02-testing.md
@@ -4,7 +4,7 @@ title: Testing
 
 Testing helps you write and maintain your code and guard against regressions. Testing frameworks help you with that, allowing you to describe assertions or expectations about how your code should behave. Svelte is unopinionated about which testing framework you use, so you can write unit tests, integration tests, and end-to-end tests using solutions like Vitest, Jasmine, Cypress and Playwright.
 
-## Unit/Integration-testing using Vitest
+## Unit/Integration testing using Vitest
 
 Unit tests allow you to test small isolated parts of your code. Integration tests allow you to test parts of your application to see if they work together. If you're using Vite (including via SvelteKit), we recommend using [Vitest](https://vitest.dev/).
 

--- a/documentation/docs/05-misc/02-testing.md
+++ b/documentation/docs/05-misc/02-testing.md
@@ -143,13 +143,13 @@ test('Component', () => {
 	// Instantiate the component using Svelte's `mount` API
 	const component = mount(Component, {
 		target: document.body, // `document` exists because of jsdom
-		props: { initial: 0 },
+		props: { initial: 0 }
 	});
 
 	expect(document.body.innerHTML).toBe('<button>0</button>');
 
 	// Click the button, then flush the changes so you can synchronously write expectations
-	document.body.querySelector('button')!.click();
+	document.body.querySelector('button').click();
 	flushSync();
 
 	expect(document.body.innerHTML).toBe('<button>1</button>');
@@ -157,7 +157,6 @@ test('Component', () => {
 	// Remove the component from the DOM
 	unmount(component);
 });
-
 ```
 
 While the process is very straightforward, it is also low level and somewhat brittle, as the precise structure of your component may change frequently. Tools like [@testing-library/svelte](https://testing-library.com/docs/svelte-testing-library/intro/) can help streamline your tests. The above test could be rewritten like this:

--- a/documentation/docs/05-misc/02-testing.md
+++ b/documentation/docs/05-misc/02-testing.md
@@ -68,7 +68,30 @@ test('Component', () => {
 
 ```
 
-While the process is very straightforward, it is also low level and somewhat brittle, as the precise structure of your component may change frequently. Tools like [@testing-library/svelte](https://testing-library.com/docs/svelte-testing-library/intro/) can help streamline your tests.
+While the process is very straightforward, it is also low level and somewhat brittle, as the precise structure of your component may change frequently. Tools like [@testing-library/svelte](https://testing-library.com/docs/svelte-testing-library/intro/) can help streamline your tests. The above test could be rewritten like this:
+
+```js
+/// file: component.test.js
+import { render, screen } from '@testing-library/svelte';
+import userEvent from '@testing-library/user-event';
+import { expect, test } from 'vitest';
+import Component from './Component.svelte';
+
+test('Component', async () => {
+	const user = userEvent.setup();
+	render(Component);
+
+	const button = screen.getByRole('button');
+	expect(button).toHaveTextContent(0);
+
+	await user.click(button);
+	expect(button).toHaveTextContent(1);
+});
+```
+
+When writing component tests that test two-way-bindings, context or snippet props, it's best to create a wrapper component for your specific test and interact with that. `@testing-library/svelte` contains some [examples](https://testing-library.com/docs/svelte-testing-library/example) on how to achieve that.
+
+> If you find yourself writing lots of component tests, think about whether you actually need to test the component, or if it's more about the logic _inside_ the component. If so, consider extracting out that logic to test it in isolation, without the overhead of a component
 
 ### Using runes inside your test files
 

--- a/documentation/docs/05-misc/02-testing.md
+++ b/documentation/docs/05-misc/02-testing.md
@@ -30,8 +30,7 @@ export default defineConfig({
 		// `// @vitest-environment jsdom` comment at the top of the test files instead.
 		environment: 'jsdom'
 	},
-	// Tell Vitest to prefer the browser condition. Svelte uses a so-called exports map
-	// which loads different code depending on the environment - we want to load the browser environment.
+	// Tell Vitest to use the `browser` entry points in `package.json` files, even though it's running in Node
 	resolve: process.env.VITEST
 		? {
 				conditions: ['browser']
@@ -40,7 +39,7 @@ export default defineConfig({
 });
 ```
 
-After that, you can create a test file in which you import the component to test, then interact with it programmatically and write expectations about the results:
+After that, you can create a test file in which you import the component to test, interact with it programmatically and write expectations about the results:
 
 ```js
 /// file: component.test.js
@@ -49,9 +48,9 @@ import { expect, test } from 'vitest';
 import Component from './Component.svelte';
 
 test('Component', () => {
-	// Instantiate the component using Svelte's mount API
-	const comp = mount(Component, {
-		target: document.body, // document.body is available thanks to JSDOM
+	// Instantiate the component using Svelte's `mount` API
+	const component = mount(Component, {
+		target: document.body, // `document` exists because of jsdom
 		props: { initial: 0 },
 	});
 
@@ -64,12 +63,12 @@ test('Component', () => {
 	expect(document.body.innerHTML).toBe('<button>1</button>');
 
 	// Remove the component from the DOM
-	unmount(comp);
+	unmount(component);
 });
 
 ```
 
-While the process is very straightforward, it is also low level and somewhat brittle - the structure of your component might change and you then may need to adjust your query selectors more often than not. Solutions like [@testing-library/svelte](https://testing-library.com/docs/svelte-testing-library/intro/) can help you facilitate this problem.
+While the process is very straightforward, it is also low level and somewhat brittle, as the precise structure of your component may change frequently. Tools like [@testing-library/svelte](https://testing-library.com/docs/svelte-testing-library/intro/) can help streamline your tests.
 
 ### Using runes inside your test files
 
@@ -128,7 +127,7 @@ E2E (short for 'end to end') tests allow you to test your full application throu
 
 To get start with Playwright, either let you guide by [their VS Code extension](https://playwright.dev/docs/getting-started-vscode), or install it from the command line using `npm init playwright`. It is also part of the setup CLI when you run `npm create svelte`.
 
-After you've done that, you should have a `tests` folder and a playwright config. You may need to adjust that config to tell Playwright what to do before running the tests - mainly starting your application at a certain port:
+After you've done that, you should have a `tests` folder and a Playwright config. You may need to adjust that config to tell Playwright what to do before running the tests - mainly starting your application at a certain port:
 
 ```js
 /// file: playwright.config.js

--- a/documentation/docs/05-misc/02-testing.md
+++ b/documentation/docs/05-misc/02-testing.md
@@ -8,7 +8,13 @@ Testing helps you write and maintain your code and guard against regressions. Te
 
 Unit tests allow you to test small isolated parts of your code. Integration tests allow you to test parts of your application to see if they work together. We recommend using [Vitest](https://vitest.dev/) for these kinds of tests (this assumes you use Vite for bundling your application).
 
-To get started, install Vitest (`npm install vitest -D`) and JSDOM (a library that shims DOM APIs, `npm install jsdom -D`). Then adjust your `vite.config.js` such that you can simulate a browser environment for testing Svelte components on the client:
+To get started, install Vitest and JSDOM (a library that shims DOM APIs):
+
+```bash
+npm install -D vitest jsdom
+```
+
+Then adjust your `vite.config.js`:
 
 ```js
 /// file: vitest.config.js

--- a/documentation/docs/05-misc/02-testing.md
+++ b/documentation/docs/05-misc/02-testing.md
@@ -6,7 +6,7 @@ Testing helps you write and maintain your code and guard against regressions. Te
 
 ## Unit/Integration-testing using Vitest
 
-Unit tests allow you to test small isolated parts of your code. Integration tests allow you to test parts of your application to see if they work together. We recommend using [Vitest](https://vitest.dev/) for these kinds of tests (this assumes you use Vite for bundling your application).
+Unit tests allow you to test small isolated parts of your code. Integration tests allow you to test parts of your application to see if they work together. If you're using Vite (including via SvelteKit), we recommend using [Vitest](https://vitest.dev/).
 
 To get started, install Vitest and JSDOM (a library that shims DOM APIs):
 

